### PR TITLE
CustomEntitlementsComputation: Prevent posting subscriber attributes in post receipt

### PR DESCRIPTION
**Pull request by aboedo**
Prevents sending subscriber attributes when posting receipts in custom entitlements computation mode.

### CODE REVIEW

File: purchases/src/main/kotlin/com/revenuecat/purchases/common/Backend.kt [hunk 1]
```diff
@@ -198,7 +198,7 @@ internal class Backend(
             "observer_mode" to observerMode,
             "price" to receiptInfo.price,
             "currency" to receiptInfo.currency,
-            "attributes" to subscriberAttributes.takeUnless { it.isEmpty() },
+            "attributes" to subscriberAttributes.takeUnless { it.isEmpty() || appConfig.customEntitlementComputation },
```
Comment (aboedo) [inline purchases/src/main/kotlin/com/revenuecat/purchases/common/Backend.kt:201]: we actually didn't have a test for this at all, so if you removed this line, tests would pass

Comment (aboedo) [reply purchases/src/main/kotlin/com/revenuecat/purchases/common/Backend.kt:201]: I added a test case for custom entitlements computation enabled and disabled

Comment (aboedo) [reply purchases/src/main/kotlin/com/revenuecat/purchases/common/Backend.kt:201]: LOOL nevermind there's an entire file dedicated to tests of exactly this. They're just not in this file and I hadn't searched thoroughly enough.

**Review by tonidero — APPROVED** [review]
Looks good!

**Review by NachoSoto — APPROVED** [review]

**Review by vegaro — APPROVED** [review]

**Pull request merged**